### PR TITLE
Update AltGr handling to accept Ctrl+Alt, add Norwegian keyboard layout

### PR DESCRIPTION
**Pull request by alexrsagen**
At least in the Norwegian keyboard layout, Ctrl+Alt is equal to AltGr.

- Added helper functions and updated existing code and layouts to recognize Ctrl+Alt as AltGr.
- Added the Norwegian keyboard layout

Also, the right alt key is only AltGr on *some* keyboard layouts, not all layouts have an AltGr key. I haven't changed anything in this regard, but it might be useful to make that distinction in the different keyboard layouts in the future (instead of in `EventDecoder::process_keyevent`).

### CODE REVIEW

**Comment by jonathanpallant** [comment]
Thank you for the PR!

For the record, could you post a photograph here of a keyboard featuring this layout?

File: src/layouts/no105.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                    DecodedKey::Unicode('`')
+                } else if modifiers.is_altgr() {
+                    DecodedKey::Unicode('´')
+                } else {
+                    DecodedKey::Unicode('\\')
+                }
+            }
+            KeyCode::Backspace => DecodedKey::Unicode(0x08.into()),
+            KeyCode::Tab => DecodedKey::Unicode(0x09.into()),
+            KeyCode::Q => {
```
Comment (jonathanpallant) [inline src/layouts/no105.rs:131]: Isn't this the same as a US 'Q'?

**Comment by alexrsagen** [comment]
The following images show keyboards with a scandinavian (Norwegian/Danish/Swedish/Finnish?) ISO layout.

HP 803181-DX1:

23BD386E-934E-4657-BE4F-4943E2D3DDBB

Logitech MK710 (Norwegian keys are colored green):

67388E6A-3F5F-44DE-9E4E-600E033781FC

File: src/layouts/no105.rs (hunk 1 above)
Comment (alexrsagen) [reply src/layouts/no105.rs]: Removed the `KeyCode::Q` handling in 2624803.

**Review by jonathanpallant — APPROVED** [review]

**Comment by jonathanpallant** [comment]
The key to the left of the Backspace has about five symbols on it!? Can you briefly explain how this works?

**Comment by jonathanpallant** [comment]
To the left of the Number 1 key, we've got six symbols, one of which is repeated 3 times...

**Comment by alexrsagen** [comment]
Welcome to the cursed world of international keyboard layouts. Yeah, some of the symbols on those keys are for different keyboard layouts. I suggest looking at the Logitech keyboard, as it colors the Norwegian layout options in green.

For the Norwegian layout, the following applies:
- `OEM8`: `|` normally, `§` with `Shift` (not affected by caps lock). No other combinations.
- `OEM+`: `\` normally, \` (accent grave) with `Shift` (not affected by `CapsLock`), `‘` (apostrophe) with `AltGr`. This is an accent key (does not apply to backslash), meaning that `AltGr`+`OEM+` followed by `e` turns into `é` for example. No other combinations.

All applications support the accent key, but I'm not sure if the OS is responsible for handling this or the individual application. Thus, I am not sure if `pc-keyboard` should also handle this.

**Comment by jonathanpallant** [comment]
Ah, I think they are called "dead keys". Currently the user of pc-keyboard will need to observe that a dead key has been pressed, and do the right thing with the next key that comes out. We could perhaps add something so that you know when it is a dead key and when it isn't, or perhaps even do the combination automatically if you request it.

**Review by jonathanpallant — APPROVED** [review]

**Pull request merged**